org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/contentassist/antlr/ContentAssistContextFactory.java

```diff
@@ -414,23 +414,28 @@ public String getPrefix(INode prefixNode) {
 			// found
 			while (startingNode.getSyntaxErrorMessage() != null) {
 				INode nodeBefore = startingNode.getPreviousSibling();
-				if (nodeBefore.getSyntaxErrorMessage() != null) {
+				if (nodeBefore != null && nodeBefore.getSyntaxErrorMessage() != null) {
 					startingNode = nodeBefore;
 				} else {
 					break;
 				}
 			}
-			return getNodeTextUpToCompletionOffset(startingNode);
+			return getNodeTextUpToCompletionOffset(startingNode, true);
 		}
 		StringBuilder result = new StringBuilder(prefixNode.getTotalLength());
 		doComputePrefix((ICompositeNode) prefixNode, result);
 		return result.toString();
 	}
 
-	public String getNodeTextUpToCompletionOffset(INode currentNode) {
+	public String getNodeTextUpToCompletionOffset(INode currentNode, boolean pseudoPrefix) {
```

Comment:
how to proper name this? split this up into two methods. do we need a for loop like doComputePrefix in getPrefix with error case too?